src/Azure.Functions.Cli/Kubernetes/KubernetesHelpers.cs

```diff
@@ -250,6 +264,88 @@ internal static string GetKedaResources(string @namespace)
                 }
             }
 
+            if (!string.IsNullOrWhiteSpace(funcAppKeysSecretsCollectionName))
+            {
+                var appKeysSecret = GetSecret(funcAppKeysSecretsCollectionName, @namespace, funcAppKeys);
+
+                if (!await ResourceExists("secret", funcAppKeysSecretsCollectionName, @namespace))
+                {
+                    result.Insert(resourceIndex, appKeysSecret);
+                    resourceIndex++;
+                }
+
+                foreach (var deployment in deployments)
+                {
+                    deployment.Spec.Template.Spec.Containers.First().EnvFrom.Append(
+                        new ContainerEnvironmentFromV1
+                        {
+                            SecretRef = new NamedObjectV1
+                            {
+                                Name = appKeysSecret.Metadata.Name
+                            }
+                        });
+                }
+            }
+            else if (!string.IsNullOrWhiteSpace(funcAppKeysConfigMapName))
+            {
+                var appKeysConfigMap = GetConfigMap(funcAppKeysConfigMapName, @namespace, funcAppKeys);
+
+                if (!await ResourceExists("configMap", funcAppKeysConfigMapName, @namespace))
+                {
+                    result.Insert(resourceIndex, appKeysConfigMap);
+                    resourceIndex++;
+                }
+
+                foreach (var deployment in deployments)
+                {
+                    deployment.Spec.Template.Spec.Containers.First().EnvFrom.Append(
+                        new ContainerEnvironmentFromV1
+                        {
+                            ConfigMapRef = new NamedObjectV1
+                            {
+                                Name = appKeysConfigMap.Metadata.Name
+                            }
+                        });
+                }
+            }
+
+            //if function keys Secrets/configMaps needs to be mounted as volume in the function runtime container
+            if (mountFuncKeysAsContainerVolume)
+            {
+                FuncAppKeysHelper.CreateFuncAppKeysVolumeMountDeploymentResource(deployments, funcAppKeysSecretsCollectionName, funcAppKeysConfigMapName);
+            }
+            else
+            {
+                var svcActName = "function-identity-svc-act";
```

Comment:
service account name should drive from the deployment name to be tied to this particular deployment. Otherwise all function apps in a given namespace will be sharing the same identity.